src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,501 @@
+//! This module provides a clickable view for a single breadcrumb.
+
+use crate::prelude::*;
+
+use super::GLYPH_WIDTH;
+use super::HORIZONTAL_MARGIN;
+use super::VERTICAL_MARGIN;
+use super::TEXT_SIZE;
+use super::RelativePosition;
+
+use crate::graph_editor::MethodPointer;
+
+use enso_frp as frp;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::display::shape::text::glyph::system::Line;
+use ensogl::display::shape::text::glyph::system::GlyphSystem;
+use ensogl::display::Sprite;
+use ensogl::gui::component;
+use ensogl::gui::component::Animation;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+use std::f32::consts::PI;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Breadcrumb top margin.
+pub const TOP_MARGIN:f32 = 0.0;
+/// Breadcrumb left margin.
+pub const LEFT_MARGIN:f32 = 0.0;
+/// Breadcrumb right margin.
+pub const RIGHT_MARGIN     : f32 = 0.0;
+const ICON_LEFT_MARGIN     : f32 = 0.0;
+const ICON_RIGHT_MARGIN    : f32 = HORIZONTAL_MARGIN;
+const ICON_RADIUS          : f32 = 8.0;
+const ICON_SIZE            : f32 = ICON_RADIUS * 2.0;
+const ICON_RING_WIDTH      : f32 = (ICON_RADIUS/4.0) as i32 as f32;
+const ICON_ARROW_SIZE      : f32 = (ICON_RADIUS/1.5) as i32 as f32;
+const SEPARATOR_SIZE       : f32 = 8.0;
+const SEPARATOR_LINE_WIDTH : f32 = 3.0;
+/// Breadcrumb padding.
+pub const PADDING          : f32 = 1.0;
+const SEPARATOR_MARGIN     : f32 = HORIZONTAL_MARGIN;
```

Comment:
Something is wrong here - `SEPARATOR_MARGIN` has the same value as `ICON_RIGHT_MARGIN`, but visually it is MUCH bigger.

Reply:
I fixed it. The separator margin was calculated by taking into consideration the text height.